client/client.go

```diff
@@ -1279,11 +1279,13 @@ func (c *Client) setupNode() error {
 	if node.Name == "" {
 		node.Name, _ = os.Hostname()
 	}
-	// TODO(dani): Fingerprint these to handle volumes that don't exist/have bad perms.
 	if node.HostVolumes == nil {
 		if l := len(c.config.HostVolumes); l != 0 {
 			node.HostVolumes = make(map[string]*structs.ClientHostVolumeConfig, l)
 			for k, v := range c.config.HostVolumes {
+				if _, err := os.Stat(v.Path); err != nil {
+					return fmt.Errorf("failed to validate volume %s, err: %v", v.Name, err)
```

Comment:
This this prevents agent startup and may be seen by an operator panicking about why their mis-provisioned cluster isn't working we may want to be more prescriptive in our error message:

```go
if os.IsNotExist(err) {
  return fmt.Errorf("host volume %q does not exist; please create %q", k, v.Path)
} else if err != nil {
  return fmt.Errorf("failed to validate volume %s, err: %v", v.Name, err)
}
```